src/overlays/actors/ovl_Bg_Mori_Elevator/z_bg_mori_elevator.c

```diff
@@ -20,37 +34,246 @@ const ActorInit Bg_Mori_Elevator_InitVars = {
     (ActorFunc)BgMoriElevator_Update,
     NULL,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Elevator/func_808A1800.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Elevator/func_808A18FC.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(uncullZoneForward, 2000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 500, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 3000, ICHAIN_CONTINUE),
+    ICHAIN_VEC3F_DIV1000(scale, 1000, ICHAIN_STOP),
+};
+
+extern UNK_TYPE D_060035F8;
+extern Gfx D_06002AD0[];
+
+f32 func_808A1800(f32* posY, f32 target, f32 arg2, f32 yVel, f32 arg4) {
+    f32 phi_f2;
+
+    phi_f2 = (target - *posY) * arg2;
+    if (*posY < target) {
+        if (yVel < phi_f2) {
+            phi_f2 = yVel;
+        } else {
+            if (phi_f2 < arg4) {
+                phi_f2 = arg4;
+            }
+        }
+        *posY = (*posY + phi_f2);
+
+        if (target < *posY) {
+            *posY = target;
+        }
+    } else {
+        if (target < *posY) {
+            if (phi_f2 < (-yVel)) {
+                phi_f2 = (-yVel);
+            } else {
+                if ((-arg4) < phi_f2) {
+                    phi_f2 = (-arg4);
+                }
+            }
+            *posY = (*posY + phi_f2);
+            if (*posY < target) {
+                *posY = target;
+            }
+        } else {
+            phi_f2 = 0.0f;
+        }
+    }
+    return phi_f2;
+}
+
+void func_808A18FC(BgMoriElevator* this, f32 distTo) {
+    f32 phi_f12;
+    f32 temp_f2;
+    f32 temp1;
+
+    // clang-format off
+    temp_f2 = fabsf(distTo) * 0.09f; 
+    if (temp_f2 < 0.0f) {
+        phi_f12 = 0.0f; } else { temp1 = (temp_f2 > 1.0f) ? (1.0f) : (temp_f2);phi_f12 = temp1;} func_800F436C(&this->dyna.actor.projectedPos, 0x2079, phi_f12);
+    // clang-format on
```

Comment:
uhhhh wtf lmao
this looks like a permuter solution to me. is there anything you can do to make this a little bit better?

id also only put the format guards around the exact thing that has to be on one line

Reply:
Yeah.  I played with it for a while and thats all that worked

Reply:
ok, well thats a shame
btw the 0x2079 in that function call looks like a sfx id